### PR TITLE
KOGITO-6441: Responsive dropdown component (OpenShift deployments)

### PR DESCRIPTION
**Pull request by thiagoelg**
# JIRA Issue
https://issues.redhat.com/browse/KOGITO-6441

# On this PR
- Added a new component to be used as a Dropdown. It wraps Patternfly dropdown components, keeping the same API while providing a way to switch to a Modal when a certain breakpoint is reached while resizing the viewport.

## Before

https://user-images.githubusercontent.com/10515135/148110845-a23e3948-5217-4dff-b3c1-1b60973079ad.mp4

## After

https://user-images.githubusercontent.com/10515135/148110860-993a362b-0815-47bc-ba10-8d1253c8fcba.mp4

### CODE REVIEW

File: packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownModal.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      isOpen={isOpen ?? false}
+      onClose={onClose}
+      hasNoBodyWrapper={true}
+      variant={ModalVariant.small}
+      title={title}
+      aria-label={title}
+      className="kogito--editor__responsive-dropdown-modal"
+    >
+      <div
+        style={{ margin: "var(--pf-global--spacer--lg) 0 0 0" }}
```
Comment (tiagobento) [inline packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownModal.tsx]: Could this style be included as part of `kogito--editor__responsive-dropdown-container`?

File: packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export function ResponsiveDropdownToggle(props: DropdownToggleProps) {
+  const { isModal } = useResponsiveDropdownContext();
+  return (
+    <>
+      {isModal ? (
+        <Button
+          variant={ButtonVariant.plain}
+          onClick={() => props.onToggle?.(props.isOpen || false)}
+          className={props.className}
+          key="bla"
```
Comment (tiagobento) [inline packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownToggle.tsx]: 🙈

File: packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const width = useWindowWidth();
+
+  return width >= breakpoints[targetBreakpoint];
+}
+
+export function useIsBelowBreakpoint(targetBreakpoint: Breakpoint) {
+  const width = useWindowWidth();
+
+  return width < breakpoints[targetBreakpoint];
+}
```
Comment (tiagobento) [inline packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts]: I feel like these two could be one single hook, like:

```typescript
export function useWindowSizeRelationToSizeBreakpoint(breakpoint: Breakpoint): { isAbove: boolean, isBelow: boolean }
```

then you could use it like this:

```typescript
const shouldBeModal = useWindowSizeRelationToSizeBreakpoint("sm").isBelow;
```

File: packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return;
+      }
+
+      ready = false;
+      func(...args);
+      setTimeout(() => {
+        ready = true;
+      }, timeout);
+    };
+  }, []);
```
Comment (tiagobento) [inline packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts]: Does this need to be a `useCallback`? Why not simply an un-exported function inside this file?

**Review by tiagobento — CHANGES_REQUESTED** [review]
Great work! Looks very good. Left some comments inline.

File: packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts (hunk 3 above)
Comment (thiagoelg) [reply packages/online-editor/src/editor/ResponsiveDropdown/hooks.ts]: Good idea!

File: packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownToggle.tsx (hunk 2 above)
Comment (thiagoelg) [reply packages/online-editor/src/editor/ResponsiveDropdown/ResponsiveDropdownToggle.tsx]: Oops

**Review by tiagobento — APPROVED** [review]

**Comment by tiagobento** [comment]
@tomasdavidorg This is ready for review.

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me.

**Pull request merged**